artemis-commons/src/test/java/org/apache/activemq/artemis/utils/ActiveMQScheduledComponentTest.java

```diff
@@ -165,4 +165,37 @@ public void run() {
       }
    }
 
+   @Test
+   public void testUsingCustomInitialDelay() throws InterruptedException {
+      final CountDownLatch latch = new CountDownLatch(1);
+      final long initialDelayMillis = 100;
+      final long checkPeriodMillis = 100 * initialDelayMillis;
+      final ActiveMQScheduledComponent local = new ActiveMQScheduledComponent(scheduledExecutorService, executorService, initialDelayMillis, checkPeriodMillis, TimeUnit.MILLISECONDS, false) {
+         @Override
+         public void run() {
+            latch.countDown();
+         }
+      };
+      final long start = System.nanoTime();
+      local.start();
+      try {
+         final boolean triggeredBeforePeriod = latch.await(local.getPeriod(), local.getTimeUnit());
+         final long timeToFirstTrigger = TimeUnit.NANOSECONDS.convert(System.nanoTime() - start, local.getTimeUnit());
```

Comment:
do you actually use NANOSECONDS on the Lock?

Reply:
Depends on the case but there are parts where I've used it: It is immune to ntpd/clock changes...

Reply:
where you need NANOSECONDS? scheduling at nanosecond may kill perf.. where is it?

Reply:
I'm not scheduling at nanos but measuring elapsed time in nanoseconds, it is different :)
The scheduling is in seconds or milliseconds in some tests.